elastic/apm-server
# Update default span/transaction type for OTel

## Motivation/summary

Align OpenTelemetry span type/subtype with the agent bridge spec:

- default to `app.internal` for internal spans
- default to `unknown` for non-internal spans
- default to `unknown` for transactions
- remove the special `${component}` attribute handling

## Checklist

~- [ ] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/master/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)~
~- [ ] Documentation has been updated~

## How to test these changes

1. Send some custom internal OTLP spans, check they get type "app", subtype "internal".
2. Send some custom non-internal OTLP spans, check they get type "unknown"

## Related issues

Closes https://github.com/elastic/apm-server/issues/6543
